Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -100,99 +98,83 @@ namespace ROS2
         // Apply transform as per origin
         PrefabMakerUtils::SetEntityTransformLocal(visual->RawPose(), entityId);
 
-        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
         auto geometry = visual->Geom();
         switch (geometry->Type())
         {
         case sdf::GeometryType::SPHERE:
             {
                 auto sphereGeometry = geometry->SphereShape();
                 AZ_Assert(sphereGeometry, "geometry is not Sphere");
-                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
-                entity->Activate();
-                LmbrCentral::SphereShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->Radius());
-                LmbrCentral::EditorShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::EditorShapeComponentRequests::SetVisibleInGame, true);
-                entity->Deactivate();
+                // Convert radius to diameter: the `_sphere_1x1.fbx.azmodel` model has a diameter of 1
+                const AZ::Vector3 sphereDimensions(sphereGeometry->Radius() * 2);
```

Comment:
This suggestion isn't really necessary, because the compiler *should* convert the int to a float at compile time, but just in case, it's generally good practice to use a float constant.
```suggestion
                const AZ::Vector3 sphereDimensions(sphereGeometry->Radius() * 2.0f);
```